code/go/0chain.net/chaincore/block/handler_test.go

```diff
@@ -0,0 +1,114 @@
+package block
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"0chain.net/core/common"
+	"0chain.net/core/datastore"
+	mocks "0chain.net/mocks/core/datastore"
+)
+
+func init() {
+	SetupHandlers()
+}
+
+func TestGetBlock(t *testing.T) {
+	type args struct {
+		ctx context.Context
+		r   *http.Request
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    interface{}
+		wantErr bool
+	}{
+		{
+			name:    "TRUE",
+			args:    args{ctx: context.TODO(), r: httptest.NewRequest(http.MethodGet, "/", nil)},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := GetBlock(tt.args.ctx, tt.args.r)
```

Comment:
These tests should be done in parallel as other errors might be missed. In practice we can expect 0chain to handle multiple HTTP test at once, using t.Parallel is an easy way to simulate that
Also, I recommend doing each test twice to help trigger any existing race issues.